### PR TITLE
Jesse/monitor 2

### PR DESCRIPTION
**Pull request by gatsbyz**
new block / tx views in explorer mode
rearranging views for more compact view

Screenshot 2023-12-19 at 11 11 07 AM

### CODE REVIEW

**Review by leovct — COMMENTED** [review]
I just took a quick look at the code and tested it but lgtm @gatsbyz

Just a couple of things:
- when scrolling down to the first block of the chain and scrolling down again, it crashes.
Screenshot 2023-12-20 at 09 56 46
- We have a lot more space in the block view (list of transactions), I think we should add the `calldata` field. This would be great for debugging.
- On top of that, being able to copy whatever value is displayed in the terminal would be awesome too!

**Comment by praetoriansentry** [comment]
Some tactical comments:
- I think some debouncing would be nice. Holding down the arrow key is a little laggy
- The timestamp column just sames `TIME`. It's probably not useful since they should all match the block right?
- It might be helpful to include / hard code the type 10 4byte selector

File: log.txt
Comment (leovct) [inline log.txt:1]: Is it a mistake?

**Review by leovct — DISMISSED** [review]
lgtm once again :)
tested against a local chain with a few hundred blocks

Couple of questions:
- Why do we need to press `g` twice to get to the top block?
- When entering monitor mode or pressing escape, no block is selected, so the `last transactions` grid doesn't contain the headlines. It only appears when a block is selected, which is a little odd imho.

Screenshot 2024-01-02 at 10 52 44

**Review by leovct — APPROVED** [review]

**Review by leovct — APPROVED** [review]

**Comment by gatsbyz** [comment]
@leovct and I discussed the empty table during initial boot up. i think it needs to wait for the transactions to load. this will be fixed in the next iteration.

**Pull request merged**